### DISCUSSION TITLE
How to resolve toml naming incompatibility with env

### DISCUSSION THREAD

**Discussion by ericedem** [spf13/viper#1351] [Q&A] [answered]
Hey, thanks for the great library!

My question revolves around a legacy app that uses `.toml` where I want to also be able to override defaults with environment variables. For example, I may have a toml file like this:

```toml
[feature]
  is_enabled = true
```

And then in golang I would do something like this:

```go
isEnabled := viper.GetBool("feature.is_enabled")
```

What I'd like to do is add an environment variable override using `viper.AutomaticEnv()` or `viper.BindEnv("feature.is_enabled")`. However it looks like what ends up happening in this case is that `"feature.is_enabled"` will be converted to an environment variable key `"FEATURE.IS_ENABLED"` which is a malformed environment key; at least in bash.

To work around this, I've manually added overrides like so:

```go
viper.BindEnv("feature.is_enabled", "FEATURE_IS_ENABLED")
```

But what I'd prefer to do is use `viper.AutomaticEnv()` so that I can override as I please without having to make new code changes or forgetting to bind some parameter. Does anyone have any recommendations here? I was thinking about opening up an issue to find and replace `.` characters in [`viper.BindEnv()`](https://github.com/spf13/viper/blob/65293ecec2f2d7237de9261f637c42d451f94a3d/viper.go#L1184) but figured asking a question first is better in case someone had an idea.

Thanks!

**Comment by sagikazarmark** [comment, marked as answer]
Take a look at the `SetEnvKeyReplacer` function.

**Reply by ericedem** [reply]
This is perfect thank you. I can do:

```
viper.AutomaticEnv()
viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
```
